src/App.js

```diff
@@ -101,10 +101,95 @@ class Board extends React.Component {
     return true;
   }
 
+  //Remove a random number from the board
+  removeRandomNumber(valueBoard) {
+
+    //Pick a random position o the board
+    let randomPositionOfBoard = Math.floor(Math.random() * valueBoard.length);
+
+    let backup = [];
+
+    //Change the random number if the position it's on was a zero
+    while(parseInt(valueBoard[randomPositionOfBoard]) === 0) {
+      //Reset random number
+      randomPositionOfBoard = Math.floor(Math.random() * valueBoard.length);
+    }
+
+    //Back up position of board's value removed and it's number
+    backup.push(randomPositionOfBoard);
```

Comment:
Maybe you can do something like

`backup.push([randomPositionOfBoard, valueBoard[randomPositionOfBoard]])`

and then backup would be a bunch of size 2 lists, one for each pair.

Reply:
It's already a bunch of size 2 lists. This: backup.push(this.removeRandomNumber(valueBoard)); | push them while grouping them up. You can test this out by logging backup in safetyRemoveNumbers method, inside the while loop.

Reply:
Oh, I see. So in this funciton, `backup` means just one pair, while in the main thread `backup` means a list of all the pairs. Can you distinguish between these two names?

Reply:
Sorry for the names...............